cli/src/pcluster/resources/login_node/user_data.sh

```diff
@@ -0,0 +1,233 @@
+Content-Type: multipart/mixed; boundary="==BOUNDARY=="
+MIME-Version: 1.0
+
+--==BOUNDARY==
+Content-Type: text/cloud-boothook; charset="us-ascii"
+MIME-Version: 1.0
+
+#!/bin/bash -x
+
+which dnf 2>/dev/null; dnf=$?
+which yum 2>/dev/null; yum=$?
+
+if [ "${!dnf}" == "0" ]; then
+  echo "proxy=${DnfProxy}" >> /etc/dnf/dnf.conf
+elif [ "${!yum}" == "0" ]; then
+  echo "proxy=${YumProxy}" >> /etc/yum.conf
+else
+  echo "Not yum system"
+fi
+
+which apt-get && echo "Acquire::http::Proxy \"${AptProxy}\";" >> /etc/apt/apt.conf || echo "Not apt system"
+
+proxy=${ProxyServer}
+if [ "${!proxy}" != "NONE" ]; then
+  proxy_host=$(echo "${!proxy}" | awk -F/ '{print $3}' | cut -d: -f1)
+  proxy_port=$(echo "${!proxy}" | awk -F/ '{print $3}' | cut -d: -f2)
+  echo -e "[Boto]\nproxy = ${!proxy_host}\nproxy_port = ${!proxy_port}\n" >/etc/boto.cfg
+  cat >> /etc/profile.d/proxy.sh <<PROXY
+export http_proxy="${!proxy}"
+export https_proxy="${!proxy}"
+export no_proxy="localhost,127.0.0.1,169.254.169.254"
+export HTTP_PROXY="${!proxy}"
+export HTTPS_PROXY="${!proxy}"
+export NO_PROXY="localhost,127.0.0.1,169.254.169.254"
+PROXY
+fi
+
+--==BOUNDARY==
+Content-Type: text/cloud-config; charset=us-ascii
+MIME-Version: 1.0
+
+bootcmd:
+  # Disable multithreading using logic from https://aws.amazon.com/blogs/compute/disabling-intel-hyper-threading-technology-on-amazon-linux/
+  # thread_siblings_list contains a comma (,) or dash (-) separated list of CPU hardware threads within the same core as cpu
+  # e.g. 0-1 or 0,1
+  # cat /sys/devices/system/cpu/cpu*/topology/thread_siblings_list
+  #     | tr '-' ','       # convert hyphen (-) to comma (,), to account that some kernels and CPU architectures use a hyphen instead of a comma
+  #     | cut -s -d, -f2-  # split over comma (,) and take the right part
+  #     | tr ',' '\n'      # convert remaining comma (,) into new lines
+  #     | sort -un         # sort and unique
+  - if [ "${DisableMultiThreadingManually}" = "true" ]; then for cpunum in $(cat /sys/devices/system/cpu/cpu*/topology/thread_siblings_list | tr '-' ',' | cut -s -d, -f2- | tr ',' '\n' | sort -un); do echo 0 > /sys/devices/system/cpu/cpu$cpunum/online; done; fi
```

Comment:
I see `DisableMultiThreadingManually` is explicitly set to false in the CDK UserData interpolation. Since the user data has a size limit can we avoid having this code segment until we actually need it?

Reply:
Yep. I agree.
I was initially focused on "making it work" and just adding/fixing what strictly necessary.
But since it is now working we can start to remove unused/unnecessary parts.

Reply:
Fixed